modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -785,27 +793,39 @@ public void onError(Throwable e) {
     }
 
     /**
-     * Saves states of the Distribution Zone Manager to vault atomically in one batch.
+     * Update topology augmentation map with newly added and removed nodes.
+     *
+     * @param addedNodes Nodes that was added to a topology and should be added to zones data nodes.
+     * @param removedNodes Nodes that was removed from a topology and should be removed from zones data nodes.
+     * @param revision Revision of the event that triggered this method.
+     * @param zoneId Zone's id.
+     */
+    private void updateTopologyAugmentationMap(Set<Node> addedNodes, Set<Node> removedNodes, long revision, int zoneId) {
+        if (!addedNodes.isEmpty()) {
+            zonesState.get(zoneId).nodesToAddToDataNodes(addedNodes, revision);
+        }
+
+        if (!removedNodes.isEmpty()) {
+            zonesState.get(zoneId).nodesToRemoveFromDataNodes(removedNodes, revision);
+        }
+    }
+
+    /**
+     * Saves states of the Distribution Zone Manager to Meta Storage atomically in one batch.
      * After restart it could be used to restore these fields.
      *
-     * @param newLogicalTopology Logical topology.
-     * @param revision Revision of the event, which triggers update.
      * @param zoneIds Set of zone id's, whose states will be staved in the Vault
      */
-    private void saveStatesToVault(Set<NodeWithAttributes> newLogicalTopology, long revision, Set<Integer> zoneIds) {
-        Map<ByteArray, byte[]> batch = IgniteUtils.newHashMap(3 + zoneIds.size());
+    private CompletableFuture<Void> saveStatesToMetastorage(Set<Integer> zoneIds) {
+        Map<ByteArray, byte[]> batch = IgniteUtils.newHashMap(1 + zoneIds.size());
 
-        batch.put(zonesLogicalTopologyVault(), toBytes(newLogicalTopology));
-
-        batch.put(zonesNodesAttributesVault(), toBytes(nodesAttributes()));
+        batch.put(zonesNodesAttributes(), toBytes(nodesAttributes()));
 
         for (Integer zoneId : zoneIds) {
-            batch.put(zoneTopologyAugmentationVault(zoneId), toBytes(zonesState.get(zoneId).topologyAugmentationMap));
+            batch.put(zoneTopologyAugmentation(zoneId), toBytes(zonesState.get(zoneId).topologyAugmentationMap));
         }
 
-        batch.put(zonesGlobalStateRevision(), longToBytes(revision));
-
-        vaultMgr.putAll(batch).join();
+        return metaStorageManager.putAll(batch);
```

Comment:
Why we don't use invoke with revision as it was described in the ticket? Seems we need to use an update revision in a condition of a ms invoke.